### PR TITLE
Add: task guide for zero shot object detection

### PR DESCRIPTION
**Pull request by MKhalusova**
This PR adds a task guide for zero-shot object detection. Unlike other task guides, there is no fine-tuning and complex preprocessing of custom data. However, the task illustrates different ways of inferencing with OWL-ViT, such as using the pipeline, manual inference with text queries, manual inference for a batch of examples, and image-guided object detection. The task guide is based on [this notebook](https://github.com/huggingface/notebooks/blob/main/examples/zeroshot_object_detection_with_owlvit.ipynb) with some additions (e.g. pipeline example) and some modifications.

### CODE REVIEW

**Comment by MKhalusova** [comment]
Images are in this PR https://huggingface.co/datasets/huggingface/documentation-images/discussions/49

File: docs/source/en/tasks/zero_shot_object_detection.mdx [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
+-->
+
+# Zero-shot object detection
+
+[[open-in-colab]]
+
+Traditionally, models used for the [object detection](object_detection) task require labeled image datasets for training,
```
Comment (stevhliu) [inline docs/source/en/tasks/zero_shot_object_detection.mdx:17]:
```suggestion
Traditionally, models used for [object detection](object_detection) require labeled image datasets for training,
```

File: docs/source/en/tasks/zero_shot_object_detection.mdx [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+and are limited to detecting the set of classes from the training data.
+
+Zero-shot object detection task supported by the [OWL-ViT](./model_doc/owlvit) model uses a different approach. OWL-ViT
```
Comment (stevhliu) [inline docs/source/en/tasks/zero_shot_object_detection.mdx:20]:
```suggestion
Zero-shot object detection is supported by the [OWL-ViT](./model_doc/owlvit) model which uses a different approach. OWL-ViT
```

File: docs/source/en/tasks/zero_shot_object_detection.mdx [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+is an open-vocabulary object detector. It means that it can detect objects in images based on free-text queries without
+the need to fine-tune the model on labeled datasets.
+
+OWL-ViT leverages multi-modal representations to perform open-vocabulary detection. It combines [CLIP](./model_doc/clip) with
+a [ViT](./model_doc/vit)-like Transformer to get multi-modal visual and text features. CLIP learns to
+associate images and their corresponding textual descriptions, and ViT can process image patches as inputs. The authors
```
Comment (stevhliu) [inline docs/source/en/tasks/zero_shot_object_detection.mdx:26]:
```suggestion
associate images and their corresponding textual descriptions, and ViT processes image patches as inputs. The authors
```

File: docs/source/en/tasks/zero_shot_object_detection.mdx [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+
+Use the processor to prepare the inputs for the model. The processor combines an image processor that prepares the
+image for the model by resizing and normalizing it, and a [`CLIPTokenizer`] that takes care of the text inputs.
+
+```py
+>>> text_queries = ["hat", "book", "sunglasses"]
+>>> inputs = processor(text=text_queries, images=im, return_tensors="pt")
+```
+
+Pass the inputs through the model, post-process, and visualize the results. As the image processor resized images before
```
Comment (stevhliu) [inline docs/source/en/tasks/zero_shot_object_detection.mdx:162]:
```suggestion
Pass the inputs through the model, post-process, and visualize the results. Since the image processor resized images before
```

File: docs/source/en/tasks/zero_shot_object_detection.mdx [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+feeding them to the model, you need to use the `post_process_object_detection` method to make sure the predicted bounding
+boxes have the correct coordinates related to the original image:
```
Comment (stevhliu) [inline docs/source/en/tasks/zero_shot_object_detection.mdx:164]:
```suggestion
boxes have the correct coordinates relative to the original image:
```

File: docs/source/en/tasks/zero_shot_object_detection.mdx [hunk 6]
```diff
@@ -0,0 +1,293 @@
...
+     <img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/transformers/tasks/zero-sh-obj-detection_5.png" alt="Cats"/>
+</div>
+
+In the preprocessing step, instead of text queries, you now need to use `query_images`:
+
+```py
+>>> inputs = processor(images=image_target, query_images=query_image, return_tensors="pt")
+```
+
+For predictions, instead of passing the inputs to model, pass them to `model.image_guided_detection()`. Draw the predictions
```
Comment (stevhliu) [inline docs/source/en/tasks/zero_shot_object_detection.mdx:270]:
```suggestion
For predictions, instead of passing the inputs to the model, pass them to [`~OwlViTForObjectDetection.image_guided_detection`]. Draw the predictions
```

File: docs/source/en/tasks/zero_shot_object_detection.mdx [hunk 7]
```diff
@@ -0,0 +1,293 @@
...
+Use the processor to prepare the inputs for the model. The processor combines an image processor that prepares the
+image for the model by resizing and normalizing it, and a [`CLIPTokenizer`] that takes care of the text inputs.
+
+```py
+>>> text_queries = ["hat", "book", "sunglasses"]
+>>> inputs = processor(text=text_queries, images=im, return_tensors="pt")
+```
+
+Pass the inputs through the model, post-process, and visualize the results. As the image processor resized images before
+feeding them to the model, you need to use the `post_process_object_detection` method to make sure the predicted bounding
```
Comment (stevhliu) [inline docs/source/en/tasks/zero_shot_object_detection.mdx:163]:
```suggestion
feeding them to the model, you need to use the [`~OwlViTImageProcessor.post_process_object_detection`] method to make sure the predicted bounding
```

File: docs/source/en/tasks/zero_shot_object_detection.mdx [hunk 8]
```diff
@@ -0,0 +1,293 @@
...
+<div class="flex justify-center">
+     <img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/transformers/tasks/zero-sh-obj-detection_4.png" alt="Beach photo with detected objects"/>
+</div>
+
+## Batch processing
+
+You can pass multiple sets of images and text queries to search for different (or same) objects in several images.
+Let's use both an astronaut image and the beach image together.
+For batch processing, you should pass text queries as a nested list to the processor and images as lists of PIL images,
+or PyTorch tensors, or NumPy arrays.
```
Comment (stevhliu) [inline docs/source/en/tasks/zero_shot_object_detection.mdx:197]:
```suggestion
PyTorch tensors, or NumPy arrays.
```

**Review by stevhliu — APPROVED** [review]
Nice work, thanks for adding this! I especially like the brief intro to the OWL-ViT model. Maybe we can embed one of the OWL-ViT demos (like this [one](https://huggingface.co/spaces/adirik/OWL-ViT)) directly on the page so users can play with it?

File: docs/source/en/tasks/zero_shot_object_detection.mdx [hunk 9]
```diff
@@ -0,0 +1,293 @@
...
+
+Traditionally, models used for the [object detection](object_detection) task require labeled image datasets for training,
+and are limited to detecting the set of classes from the training data.
+
+Zero-shot object detection task supported by the [OWL-ViT](./model_doc/owlvit) model uses a different approach. OWL-ViT
+is an open-vocabulary object detector. It means that it can detect objects in images based on free-text queries without
+the need to fine-tune the model on labeled datasets.
+
+OWL-ViT leverages multi-modal representations to perform open-vocabulary detection. It combines [CLIP](./model_doc/clip) with
+a [ViT](./model_doc/vit)-like Transformer to get multi-modal visual and text features. CLIP learns to
```
Comment (alaradirik) [inline docs/source/en/tasks/zero_shot_object_detection.mdx:25]: ViT is used as the vision backbone of CLIP, this might make it a bit clearer:
```suggestion
OWL-ViT leverages multi-modal representations to perform open-vocabulary detection. It combines [CLIP](./model_doc/clip) with
lightweight object classification and localization heads. Open-vocabulary detection is achieved by embedding free-text queries with the text encoder of CLIP and using them as input to the object classification and localization heads.
```

File: docs/source/en/tasks/zero_shot_object_detection.mdx [hunk 10]
```diff
@@ -0,0 +1,293 @@
...
+
+>>> image
+```
+
+<div class="flex justify-center">
+     <img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/transformers/tasks/zero-sh-obj-detection_1.png" alt="Astronaut Eileen Collins"/>
+</div>
+
+Pass the image and the candidate object labels to look for to the pipeline.
+Here we pass the image directly; other suitable options include a local path to an image or an image url.
```
Comment (alaradirik) [inline docs/source/en/tasks/zero_shot_object_detection.mdx:74]: Could you extend this a bit to describe what type of data is passed to `candidate_labels`?

```suggestion
Here we pass the image directly; other suitable options include a local path to an image or an image url. We also pass text descriptions for all items we want to query the image for. 
```

**Review by alaradirik — APPROVED** [review]
I left a few comments but looks great to me! Thanks for adding this :)

**Comment by MKhalusova** [comment]
> Nice work, thanks for adding this! I especially like the brief intro to the OWL-ViT model. Maybe we can embed one of the OWL-ViT demos (like this [one](https://huggingface.co/spaces/adirik/OWL-ViT)) directly on the page so users can play with it?

Thanks for the suggestion! I embedded demo :)

File: docs/source/en/tasks/zero_shot_object_detection.mdx [hunk 11]
```diff
@@ -0,0 +1,302 @@
...
+-->
+
+# Zero-shot object detection
+
+[[open-in-colab]]
+
+Traditionally, models used for [object detection](object_detection) require labeled image datasets for training,
+and are limited to detecting the set of classes from the training data.
+
+Zero-shot object detection is supported by the [OWL-ViT](./model_doc/owlvit) model which uses a different approach. OWL-ViT
```
Comment (sgugger) [inline docs/source/en/tasks/zero_shot_object_detection.mdx:20]:
```suggestion
Zero-shot object detection is supported by the [OWL-ViT](../model_doc/owlvit) model which uses a different approach. OWL-ViT
```
You need to get out of the tasks subfolder.

File: docs/source/en/tasks/zero_shot_object_detection.mdx [hunk 12]
```diff
@@ -0,0 +1,302 @@
...
+is an open-vocabulary object detector. It means that it can detect objects in images based on free-text queries without
+the need to fine-tune the model on labeled datasets.
+
+OWL-ViT leverages multi-modal representations to perform open-vocabulary detection. It combines [CLIP](./model_doc/clip) with
```
Comment (sgugger) [inline docs/source/en/tasks/zero_shot_object_detection.mdx:24]:
```suggestion
OWL-ViT leverages multi-modal representations to perform open-vocabulary detection. It combines [CLIP](../model_doc/clip) with
```

File: docs/source/en/tasks/zero_shot_object_detection.mdx [hunk 13]
```diff
@@ -0,0 +1,302 @@
...
+
+Before you begin, make sure you have all the necessary libraries installed:
+
+```bash
+pip install -q transformers
+```
+
+## Zero-shot object detection pipeline
+
+The simplest way to try out inference with OWL-ViT is to use it in a [`Pipeline`]. Instantiate a pipeline
```
Comment (sgugger) [inline docs/source/en/tasks/zero_shot_object_detection.mdx:45]:
```suggestion
The simplest way to try out inference with OWL-ViT is to use it in a [`pipeline`]. Instantiate a pipeline
```

**Review by sgugger — APPROVED** [review]
Thanks a lot for writing this guide!
